src/AgentVisibilityContext.tsx

```diff
@@ -0,0 +1,141 @@
+import React, { createContext } from 'react';
+import {
+  DisplayAction,
+  DisplayStateEntry,
+  HiddenOrHighlightedState,
+  SubAgentDisplayMaps,
+  TopLevelDisplayStatus,
+  UIDisplayEntry,
+  ViewerVisibilityMap,
+  ViewerVisibilityStates,
+} from './constants';
+import {
+  getPayloadSubAgentDisplayAction,
+  getPayloadTopLevelDisplayAction,
+  getSubAgentDisplayStateMap,
+  getTopLevelDisplayStatus,
+} from './selectors';
+
+interface VisibilityContextType {
+  currentVisibilityStates: ViewerVisibilityStates;
+  updateCurrentVisibilityStates: (
+    newValue: ViewerVisibilityMap,
+    hideOrHighlight: DisplayAction
+  ) => void;
+  handleSubAgentDisplayAction: (
+    agent: UIDisplayEntry,
+    hideOrHighlight: DisplayAction,
+    subAgent?: DisplayStateEntry
+  ) => void;
+  handleDisplayActionTopLevel: (
+    agent: UIDisplayEntry,
+    hideOrHighlight: DisplayAction
+  ) => void;
+  updateTopLevelDisplayStatus: (agent: UIDisplayEntry) => TopLevelDisplayStatus;
+  updateSubAgentDisplayMaps: (agent: UIDisplayEntry) => SubAgentDisplayMaps;
+}
+const { Hide, Highlight } = DisplayAction;
+const { Inactive } = HiddenOrHighlightedState;
+
+export const VisibilityContext = createContext<VisibilityContextType>({
+  currentVisibilityStates: { hidden: {}, highlight: {} },
+  updateCurrentVisibilityStates: () => {},
+  handleSubAgentDisplayAction: () => {},
+  handleDisplayActionTopLevel: () => {},
+  updateTopLevelDisplayStatus: () => {
+    return { hidden: Inactive, highlight: Inactive };
+  },
+  updateSubAgentDisplayMaps: () => {
+    return { hidden: {}, highlight: {} };
+  },
+});
```

Comment:
This initial const declaration are the default values that are a failsafe if there is not context provider in the tree above a component that calls on the context, but there should always be a provider.